Documentation/botr/mixed-mode.md

```diff
@@ -0,0 +1,36 @@
+# Mixed Mode Assemblies
+## Introduction
+Most interoperability between managed and native code uses P/Invokes, COM, or WinRT. Since P/Invokes and COM are bound to native code at runtime, they’re susceptible to mistakes ranging from incorrect naming to subtle mistakes in signatures that cause stack corruption. COM can also be used to call from native to managed code, but it requires a significant amount of setup. WinRT avoids those problems but isn’t available in all cases.
+
+C++/CLI provides a different, compiler-verified approach to interoperability called mixed-mode assemblies (also sometimes referred to as It-Just-Works or IJW). Rather than requiring that developers do special declarations like P/Invokes, the C++ compiler automatically generates everything needed to transition to and from managed and native code. Additionally, the C++ compiler decides whether a given C++ method should be managed or native, so even within an assembly, transitions happen regularly and without developer intervention required.
+
+## Calling Native Code
+C++/CLI code may call into either native code in the same assembly or a different library. Calls to a different library generates P/Invokes similar to those that might be written by hand in C#. However, calls to the same assembly work differently. While P/Invokes to different libraries specify the name of the library and the name of an export to call, P/Invokes to the same library have a null entry point and set an RVA – an address within the library to call. In metadata, that looks like:
```

Comment:
Could you add a little more info about how C++/CLI provides compiler-verified(Static-type checking) for call to a different library? header file?

Reply:
Added